lib/seahorn/HornClauseDBTransf.cc

```diff
@@ -58,4 +58,80 @@ void normalizeHornClauseHeads(HornClauseDB &db) {
     db.addRule(new_rule);
   }
 }
+
+template <typename Set, typename Predicate> void erase_if(Set &s, Predicate shouldRemove) {
+  for(auto it = s.begin(); it!=s.end();){
+    if(shouldRemove(*it)){
+      it = s.erase(it);
+    }
+    else {
+      ++it;
+    }
+  }
+}
+
+// -- tdb is an empty db that will contain db after transformation
+void removeFiniteMapsHornClausesTransf(HornClauseDB &db, HornClauseDB &tdb) {
+
+  ExprFactory &efac = tdb.getExprFactory();
+  ExprMap predDeclTransf;
+
+  // Remove Finite Maps arguments
+  for (auto predIt : db.getRelations()) {
+    Expr predDecl = predIt;
+
+    Expr newPredDecl;
+    if (predDecl->arity() < 2) // just return type?, is this assumption correct?
+      newPredDecl = predDecl;
+    else{
```

Comment:
formatting. was `clang-format` ran on this at all? I saw other formatting issues but assumed it was due to github formatting.

Reply:
I thought emacs was doing it already but I ran `clang-format` everywhere now